### PR TITLE
Show comment/requests in report preview

### PR DESCRIPTION
**Pull request by Gonals**
### Details
cc @JmillsExpensify, @trjExpensify  
### Fixed Issues

$ https://github.com/Expensify/App/issues/21178

### Tests
1. Log into your account in newdot
2. Request money from someone else and add a comment while doing so
3. Confirm the comment immediately shows in the request:
Screenshot 2023-07-26 at 4 02 53 PM

4. Request money again from the same person.
5. Confirm that, instead of the comment, the number of requests show immediately:
Screenshot 2023-07-26 at 4 02 59 PM

- [x] Verify that no errors appear in the JS console

### Offline tests
Same as tests but offline

### QA Steps
Same as tests

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

Screenshot 2023-07-26 at 4 03 16 PM

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

Screenshot 2023-07-26 at 4 48 47 PM

</details>

<details>
<summary>Desktop</summary>

Screenshot 2023-07-26 at 4 09 18 PM

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

</details>

### CODE REVIEW

**Comment by Gonals** [comment]
Yes, this should be ready for review

**Comment by Gonals** [comment]
@abdulrahuman5196 bump!

**Comment by abdulrahuman5196** [comment]
Sorry for the delay. Will review in couple of hours.

**Comment by abdulrahuman5196** [comment]
@Gonals 
Request count is not updated in the preview if a request is deleted, the preview is only updated when the request is added. I assume it has to do with backend(not sure though). should that be a separate PR?


https://github.com/Expensify/App/assets/46707890/02e79bff-b3c9-41f2-98a4-35181bf43315

**Comment by Gonals** [comment]
> @Gonals Request count is not updated in the preview if a request is deleted, the preview is only updated when the request is added. I assume it has to do with backend(not sure though).
> 
>  Screen.Recording.2023-08-02.at.9.35.02.PM.mov

Nice catch! it is 50-50. 

@abdulrahuman5196 I have fixed the frontend here to optimistically update it, but the backend bug remains. Let's ignore it for this PR and I'll fix it in a different issue. Let's see if we can merge today, as I'm OOO staring tomorrow!

**Comment by abdulrahuman5196** [comment]
@Gonals Got it. I tested now. I think the optimistically updating data is also not complete and in cases I still seeing issues.

Like
1) Description is not updating in preview if we delete all requests and keep only one request with description

But I am fine with this, given we are planning to create separate bug to address the request count not getting updating on requests deletion/updation. I think it would be best to re-test the optimistic function and backend verification as well in that issue.

**Comment by abdulrahuman5196** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>





https://github.com/Expensify/App/assets/46707890/0b9b84e7-750d-45b3-ab3e-83f63a04005c



</details>

<details>
<summary>Mobile Web - Chrome</summary>






https://github.com/Expensify/App/assets/46707890/49d1eece-7f9f-45d6-95b6-85bf40f5aa34


</details>

<details>
<summary>Mobile Web - Safari</summary>






https://github.com/Expensify/App/assets/46707890/07f610be-0f53-4ffa-a4ef-d662161e50f2


</details>

<details>
<summary>Desktop</summary>







https://github.com/Expensify/App/assets/46707890/0ad3ee17-043f-4e6f-82cc-8491f481940a



</details>

<details>
<summary>iOS</summary>






https://github.com/Expensify/App/assets/46707890/10b951f0-b7a7-44c1-ac32-4309a23a27bf


</details>

<details>
<summary>Android</summary>





https://github.com/Expensify/App/assets/46707890/afd4c5f5-6a61-439d-adb0-f6a858aa3f4e



</details>

**Comment by abdulrahuman5196** [comment]
@Gonals Changes looks good. We have to create 2 issues

1) Request count/description is not properly updated on request deletion/updation

2) Request count/description is not updated on workspace chat money requests during addition for any pre-existing workspace. But working fine for new workspaces.

https://github.com/Expensify/App/assets/46707890/4a16b7a9-fa0b-4823-861e-27bd1f3928eb

**Review by abdulrahuman5196 — APPROVED** [review]
Changes looks good and works well. Reviewers checklist is also complete.

All yours. @Gonals 

🎀 👀 🎀
C+ Reviewed

File: src/components/ReportActionItem/ReportPreview.js [hunk 1]
```diff
@@ -95,6 +95,9 @@ const defaultProps = {
 function ReportPreview(props) {
     const managerID = props.iouReport.managerID || props.action.actorAccountID || 0;
     const isCurrentUserManager = managerID === lodashGet(props.session, 'accountID');
+    const moneyRequestCount = lodashGet(props.action, 'childMoneyRequestCount', 0);
+    const moneyRequestComment = lodashGet(props.action, 'childLastMoneyRequestComment');
```
Comment (MonilBhavsar) [inline src/components/ReportActionItem/ReportPreview.js:99]: NAB, set empty string as default?

**Review by MonilBhavsar — DISMISSED** [review]

File: src/components/ReportActionItem/ReportPreview.js (hunk 1 above)
Comment (Gonals) [reply src/components/ReportActionItem/ReportPreview.js]: Done

**Review by MonilBhavsar — DISMISSED** [review]

**Comment by MonilBhavsar** [comment]
Do we need to update this function usage to pass `comment` param https://github.com/Expensify/App/blame/2335cbaf854f236db28cbbc72fa0446a6584d78c/src/libs/actions/IOU.js#L1257

File: src/libs/ReportUtils.js [hunk 2]
```diff
@@ -1852,10 +1852,11 @@ function buildOptimisticIOUReportAction(
  *
  * @param {Object} chatReport
  * @param {Object} iouReport
+ * @param {String} comment - User comment for the IOU.
  *
  * @returns {Object}
  */
-function buildOptimisticReportPreview(chatReport, iouReport) {
+function buildOptimisticReportPreview(chatReport, iouReport, comment) {
```
Comment (MonilBhavsar) [inline src/libs/ReportUtils.js:1859]: Sorry, we need to make optional param for this function too?

Comment (Gonals) [reply src/libs/ReportUtils.js]: Done!

File: src/components/ReportActionItem/ReportPreview.js [hunk 3]
```diff
@@ -95,6 +95,9 @@ const defaultProps = {
 function ReportPreview(props) {
     const managerID = props.iouReport.managerID || props.action.actorAccountID || 0;
     const isCurrentUserManager = managerID === lodashGet(props.session, 'accountID');
+    const moneyRequestCount = lodashGet(props.action, 'childMoneyRequestCount', 0);
+    const moneyRequestComment = lodashGet(props.action, 'childLastMoneyRequestComment', '');
+    const showComment = moneyRequestComment || moneyRequestCount > 1;
```
Comment (luacmartins) [inline src/components/ReportActionItem/ReportPreview.js:100]: NAB 
```suggestion
    const shouldShowComment = moneyRequestComment || moneyRequestCount > 1;
```

**Review by luacmartins — APPROVED** [review]

File: src/libs/ReportUtils.js [hunk 4]
```diff
@@ -1900,6 +1904,8 @@ function updateReportPreview(iouReport, reportPreviewAction) {
                 type: CONST.REPORT.MESSAGE.TYPE.COMMENT,
             },
         ],
+        childLastMoneyRequestComment: comment || reportPreviewAction.childLastMoneyRequestComment,
+        childMoneyRequestCount: reportPreviewAction.childMoneyRequestCount + 1,
```
Comment (eVoloshchak) [inline src/libs/ReportUtils.js:1908]: 👋 Coming from https://github.com/Expensify/App/issues/26767
We shouldn't increase `childMoneyRequestCount` if you selected "Pay Elsewhere"
More details [here](https://github.com/Expensify/App/issues/26767#issuecomment-1706737988)

**Pull request merged**